buildtest/settings/default.yml

```diff
@@ -0,0 +1,11 @@
+
+editor: vi
+executors:
+  local:
+    type: local
+    description: submit jobs locally
+  slurm:
+    type: slurm
+    description: submit jobs with slurm using srun instead of sbatch
+    vars:
```

Comment:
i think of ``vars`` where you can declare variables  that get passed in to all test that use the executor.
The launcher should be its own key part of slurm and local that should be of type `string` with `enum` to validate launcher type. Initially for slurm we can use `srun`, `mpirun`, `mpiexec`. For local it could be `mpirun`, `mpiexec`, `local`. The way i intended launcher to be used  is to figure out how job should be dispatched.  

The `local` executor should target atleast `mpirun` so that a user who wants to run MPI code in a system without scheduler can run it.

Reply:
The launcher is a slurm specific variable, so the user can define it under variables and it will be parsed given that `type` is slurm. I'm doing a simple implementation now so I'm not adding any of the mpi tools, but they could be other options for that variable as well. If there is some derivative of local that uses MPI, that's also another customization / variable (not a default).